channels/ON-CHAIN.md

```diff
@@ -363,15 +375,144 @@ channel.
 - `channel_id`
 - `initiator_pubkey`
 - `responder_pubkey`
-- `total_amount`
+- `channel_amount`
 - `initiator_amount`
+- `responder_amount`
 - `channel_reserve`
 - `state_hash`: last published state_hash 
 - `round`: last published round
-- `closes_at`: on-chain channel closing height
 - `lock_period`: agreed upon locking period by peers
+- `closes_at`: on-chain channel closing height
+- `force_blocked_until`: on-chain channel height after which a new force
+  progress can be included in a block
 
-Keeping track of the `state_hash`, `round`, `closes_at` and `lock_period` is
+Keeping track of the `state_hash`, `round`, `closes_at`, `force_blocked_until` and `lock_period` is
 necessary for nodes to be able to assess the validity of `channel_slash` and
 `channel_settle` transactions.
 Serialization defined [here](../serializations.md#channel)
+
+## Force progress on-chain
+
+Forcing progress is the mechanism to be used when a dispute
+arises between parties and one of them wants to use the blockchain as an
+arbiter. The poster provides enough state so that an off-chain contract can be
+executed on-chain and thus producing the channel's next off-chain state.
+
+This can happen both while a channel is closing or while it is still active. If the
+channel is not closing - participants can continue using it from the
+on-chain produced state or initiate a closing. If the channel is already
+closing, the force-progress updates what are the currently expected closing
+amounts for each participant (according to the contract's execution).
+
+The force progress is based on what is considered to be the latest off-chain
+state. We have no way of proving that this is the case so there is a time frame
+provided for the other participant to dispute it. This time frame is
+measured in block height and it works exactly like the `lock_period` does for
```

Comment:
Is this time frame negotiated somewhere?

Reply:
Addressed in [
d59f0cc](https://github.com/aeternity/protocol/pull/180/commits/d59f0ccf2cfb514947638dce1e6421508d2de61f#diff-4fbc1ee2b0f1c1c88eeeafc062a7989cR412)